content/en/blog/tracing-syscalls-using-ebpf/index.md

```diff
@@ -0,0 +1,278 @@
+
+- `tp/syscalls/sys_enter_execve` refers to a tracepoint that records when a process spawns the execve system call.
+
+- A list of all the available tracepoints is present in the `/sys/kernel/debug/tracing/available_events` file. The format for each line in the file is `<category>:<name>`. For example, syscalls:sys_enter_execve. 
+
+Before compiling the program, we need to do some basic configuration:
+
+<br><a target="_blank" href="images/ebpf-02.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Basic configurations."
+       src="images/ebpf-02.png">
+  </img>
+</a><br><br>
+
+Let's compile the program. 
+The following command can be used to do this task:
+
+<br><a target="_blank" href="images/ebpf-03.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Command to compile the program."
+       src="images/ebpf-03.png">
+  </img>
+</a><br><br>
+
+Now, we need to write a loader program that loads and attaches this program. This loader program is used to load and attach an eBPF program to the Linux kernel. It opens and loads the eBPF object file, checks for errors during the process, finds a specific eBPF program within the loaded object, and attaches it to the kernel. Once attached, the eBPF program will be executed when certain events occur. The program enters an infinite loop at the end, indicating that it will continue running until it is manually terminated.
+
+<br><a target="_blank" href="images/ebpf-04.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Loader program for eBPF."
+       src="images/ebpf-04.png">
+  </img>
+</a><br><br>
+
+Let's compile and run this program
+
+<br><a target="_blank" href="images/ebpf-05.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Command to compile loader program for eBPF."
+       src="images/ebpf-05.png">
+  </img>
+</a><br><br>
+
+To get the logs generated by the function `bpf_printk`, we can read the file:
+`/sys/kernel/tracing/trace_pipe`
+
+<br><a target="_blank" href="images/ebpf-06.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Reading '/sys/kernel/tracing/trace_pipe' file."
+       src="images/ebpf-06.png">
+  </img>
+</a><br><br>
+
+<br><a target="_blank" href="images/ebpf-res-01.png">
+  <img style="border: 2px solid #00b4c8" 
+       alt="Messages we got after reading the 'trace_pipe' file."
+       src="images/ebpf-res-01.png">
+  </img>
+</a><br><br>
+
+Manually reading messages from the tracepipe doesn't seem to be an efficient approach. It would be advantageous to establish a mechanism for the eBPF program to send messages to the loader program. One viable solution is to utilise ring buffers. Let’s review more details about ring buffers.
```

Comment:
```suggestion
Manually reading messages from the tracepipe doesn't seem to be an efficient approach. It would be advantageous to establish a mechanism for the eBPF program to send messages to the loader program. One viable solution is to utilize ring buffers. Let’s review more details about ring buffers.
```

Reply:
done